Remotion/Data/DomainObjects.UnitTests/Tracing/TracingDbConnectionTest.cs

```diff
@@ -161,15 +297,193 @@ public void Close ()
+
+      var assumedResult = new DataTable();
+
+      _innerConnectionMock.Setup(mock => mock.GetSchema(param, param2)).Returns(assumedResult).Verifiable();
+
+      var result = _connection.GetSchema(param, param2);
+
+      _innerConnectionMock.Verify();
+      _extensionMock.Verify();
+      Assert.That(result, Is.EqualTo(assumedResult));
+    }
+
+    [Test]
+    public void GetSchemaAsync ()
+    {
+      var token = CancellationToken.None;
+      var assumedResult = Task.FromException<DataTable>(new InvalidOperationException("Should not get called."));
+
+      _innerConnectionMock.Setup(mock => mock.GetSchemaAsync(token)).Returns(assumedResult).Verifiable();
+
+      var result = _connection.GetSchemaAsync(token);
+
+      _innerConnectionMock.Verify();
+      _extensionMock.Verify();
+      Assert.That(result, Is.EqualTo(assumedResult));
+    }
+
+    [Test]
+    public void GetSchemaAsync_WithCollectionName ()
+    {
+      var token = CancellationToken.None;
+      var param = "collection name";
+      var assumedResult = Task.FromException<DataTable>(new InvalidOperationException("Should not get called."));
+
+      _innerConnectionMock.Setup(mock => mock.GetSchemaAsync(param, token)).Returns(assumedResult).Verifiable();
+
+      var result = _connection.GetSchemaAsync(param, token);
+
+      _innerConnectionMock.Verify();
+      _extensionMock.Verify();
+      Assert.That(result, Is.EqualTo(assumedResult));
+    }
+
+    [Test]
+    public void GetSchemaAsync_WithCollectionName_AndRestrictionValues ()
+    {
+      var token = CancellationToken.None;
+      var param = "collection name";
+      var param2 = new[] { "asdf", "qwerty" };
+
+      var assumedResult = Task.FromException<DataTable>(new InvalidOperationException("Should not get called."));
+
+      _innerConnectionMock.Setup(mock => mock.GetSchemaAsync(param, param2, token)).Returns(assumedResult).Verifiable();
+
+      var result = _connection.GetSchemaAsync(param, param2, token);
+
+      _innerConnectionMock.Verify();
+      _extensionMock.Verify();
+      Assert.That(result, Is.EqualTo(assumedResult));
+    }
+
+    [Test]
+    public void CreateCommand ()
+    {
+      var commandMock = new Mock<DbCommand>(MockBehavior.Strict);
+      commandMock.Protected().Setup("Dispose", [false]); // for Finalizer
+      _innerConnectionMock.Protected().Setup<DbCommand>("CreateDbCommand").Returns(commandMock.Object).Verifiable();
+
+      var tracingDbCommand = ((TracingDbConnection)_connection).CreateCommand();
+
+      _innerConnectionMock.Verify();
+      _extensionMock.Verify();
+      commandMock.Verify();
+
+      Assert.That(tracingDbCommand.WrappedInstance, Is.EqualTo(commandMock.Object));
+      Assert.That(tracingDbCommand.PersistenceExtension, Is.EqualTo(_extensionMock.Object));
+      Assert.That(tracingDbCommand.ConnectionID, Is.EqualTo(((TracingDbConnection)_connection).ConnectionID));
+    }
+
+    [Test]
+    public void CreateCommandWithPublicImplementation ()
```

Comment:
naming consistency:
in TracingCommandTest, you uses "WithPublicImplementation" for the NEW implementation, here it's the reverse. 
Perhaps we should simply use "WithNewImplementation" (or ask @a-ctor for a good naming). The naming should be consistent between the for Tracing-test fixtures